app/common/lib/ledgerUtil.js

```diff
@@ -183,16 +186,136 @@ const stickyP = (state, publisherKey) => {
   return (result === undefined || result)
 }
 
-module.exports = {
-  shouldTrackView,
-  batToCurrencyString,
-  formattedTimeFromNow,
-  formattedDateFromTimestamp,
-  walletStatus,
-  blockedP,
-  contributeP,
-  visibleP,
-  eligibleP,
-  stickyP,
-  formatCurrentBalance
+const getMediaId = (data, type) => {
+  let id = null
+
+  if (type == null || data == null) {
+    return id
+  }
+
+  switch (type) {
+    case ledgerMediaProviders.YOUTUBE:
+      {
+        id = data.docid
+        break
+      }
+  }
+
+  return id
+}
+
+const getMediaKey = (id, type) => {
+  if (id == null || type == null) {
+    return null
+  }
+
+  return `${type.toLowerCase()}_${id}`
+}
+
+const getMediaData = (xhr, type) => {
+  let result = null
+
+  if (xhr == null || type == null) {
+    return result
+  }
+
+  switch (type) {
+    case ledgerMediaProviders.YOUTUBE:
+      {
+        const parsedUrl = urlParse(xhr)
+        let query = null
+
+        if (parsedUrl && parsedUrl.query) {
+          query = queryString.parse(parsedUrl.query)
+        }
+        result = query
+        break
+      }
+  }
+
+  return result
+}
+
+const getMediaDuration = (data, type) => {
+  let duration = 0
+  switch (type) {
+    case ledgerMediaProviders.YOUTUBE: {
+      duration = getYoutubeDuration(data)
+      break
+    }
+  }
+
+  return duration
+}
+
+const getYoutubeDuration = (data) => {
```

Comment:
nit: to respect the brand name, it should be YouTube (ex: `getYouTubeDuration`)

Reply:
fixed